net/golioth/golioth.c

```diff
@@ -90,6 +90,18 @@ static int golioth_setsockopt_dtls(struct golioth_client *client, int sock,
 		}
 	}
 
+	/* If Connection IDs are enabled, set socket option to send CIDs, but not require that the
+	 * server sends one in return.
+	 */
+#ifdef CONFIG_MBEDTLS_SSL_DTLS_CONNECTION_ID
```

Comment:
I think we should introduce new Kconfig option that will trigger whether Golioth connection will use DTLS CID or not. This option should select `CONFIG_MBEDTLS_SSL_DTLS_CONNECTION_ID` when enabled.

The main reason is that in some situation `CONFIG_MBEDTLS_SSL_DTLS_CONNECTION_ID=y` might be selected already (e.g. by another library/dependency) and in other user might want to use DTLS CID only for their purpose. Hence we should have a separate option that will control behavior of the Golioth SDK alone.

Reply:
Makes sense!